syntax = "proto3";

package protos;
option go_package = "orch-go/gen/protos;protos";
option csharp_namespace = "Core";

import "custom_types.proto";
import "google/protobuf/empty.proto";
import "user_credential.proto";

service UserCredentialAnalyticService {
  rpc ProcessGetAll (GetAllRequest) returns (GetAllUserCredentialsResponse);
  rpc ProcessGetById (GetUserCredentialByIdRequest) returns (UserCredentialModel);
  rpc ProcessGetByIds (GetUserCredentialByIdsRequest) returns (GetAllUserCredentialsResponse);
  rpc ProcessGetByUsername (GetUserCredentialByUsernameRequest) returns (UserCredentialModel);
  rpc ProcessGetCount (google.protobuf.Empty) returns (CountResponse);
}
